File: LeetCode/[E] - 1768. Merge Strings Alternately/solution.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        res = []
        p1, p2 = 0, 0
        len_w1, len_w2 = len(word1), len(word2)
        choose = 0
        while p1 != len_w1 and p2 != len_w2:
            curr_word = [word1, word2][choose]
            curr_p = [p1, p2][choose]
            res.append(curr_word[curr_p])
            if choose == 0:
                p1 += 1
            else:
                p2 += 1
            choose = (choose + 1) % 2
        if p1 == len_w1:
            res.extend(list(word2[p2:]))
        else:
            res.extend(list(word1[p1:]))
        return "".join(res)

File: LeetCode/[E] - 1768. Merge Strings Alternately/solution.go
package main

func mergeAlternately(word1 string, word2 string) string {
	var (
		p1 int
		p2 int
	)
	res := make([]byte, 0, len(word1)+len(word2))
	ch := 0
	wrds := []string{word1, word2}
	pnts := []*int{&p1, &p2}
	for p1 != len(word1) && p2 != len(word2) {
		curr_wrd := wrds[ch]
		curr_pnt := pnts[ch]
		res = append(res, curr_wrd[*curr_pnt])
		if ch == 0 {
			p1 += 1
		} else {
			p2 += 1
		}
		ch = (ch + 1) % 2
	}
	if p1 == len(word1) {
		for p2 != len(word2) {
			res = append(res, word2[p2])
			p2 += 1
		}
	} else {
		for p1 != len(word1) {
			res = append(res, word1[p1])
			p1 += 1
		}
	}
	return string(res)
}
